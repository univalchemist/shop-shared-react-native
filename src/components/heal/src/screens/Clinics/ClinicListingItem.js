import React from 'react';
import PropTypes from 'prop-types';
import { Dimensions, TouchableOpacity } from 'react-native';
import {
  Box,
  Text,
  PlainText,
  ListItemWithRightChevron,
  Image,
} from '@heal/src/wrappers/components';
import { connect } from 'react-redux';
import { useTheme, useIntl } from '@heal/src/wrappers/core/hooks';

import { iconLocation, panelDoctor } from '@components/heal/images';

const screenSize = Dimensions.get('window');

const ClinicListingItem = ({ clinic, latitude, longitude, onPress }) => {
  const intl = useIntl();
  const theme = useTheme();

  const shortenDistance = distance => {
    if (distance > 1000) {
      return intl.formatMessage(
        { id: 'heal.clinic.kmdistance' },
        {
          number: distance && (distance / 1000).toFixed(2),
        },
      );
    }

    return intl.formatMessage(
      { id: 'heal.clinic.mdistance' },
      {
        number: distance && distance.toFixed(2),
      },
    );
  };

  return (
    <TouchableOpacity onPress={onPress}>
      <ListItemWithRightChevron
        width={screenSize.width}
        height={134}
        justifyContent="center"
        paddingLeft={32}
        paddingRight={30}
      >
        <Text
          fontSize={21}
          color={theme.colors.defaultText}
          lineHeight={26}
          letterSpacing={0.3}
        >
          {clinic.name}
        </Text>
        <PlainText numberOfLines={1}>{clinic.address}</PlainText>
        <Box flexDirection="row" marginTop={16}>
          <Box flexDirection="row">
            <Image source={panelDoctor} />
            <Text
              fontSize={14}
              lineHeight={18}
              letterSpacing={0.3}
              marginLeft={8}
            >
              {clinic.doctors.length === 1
                ? intl.formatMessage(
                    { id: 'clinic.singularDoctor' },
                    { number: clinic.doctors.length },
                  )
                : intl.formatMessage(
                    { id: 'clinic.pluralDoctor' },
                    { number: clinic.doctors.length },
                  )}
            </Text>
          </Box>
          {latitude === 0 ||
          longitude === 0 ||
          clinic.distanceToClient === undefined ? null : (
            <Box flexDirection="row" marginLeft={32} marginRight={32}>
              <Image source={iconLocation} />
              <Text
                fontSize={14}
                lineHeight={18}
                letterSpacing={0.3}
                marginLeft={8}
              >
                {shortenDistance(clinic.distanceToClient)}
              </Text>
            </Box>
          )}
        </Box>
      </ListItemWithRightChevron>
    </TouchableOpacity>
  );
};

ClinicListingItem.propTypes = {
  clinic: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    doctors: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        locale: PropTypes.string,
        name: PropTypes.string,
        gender: PropTypes.string,
        isActive: PropTypes.bool,
      }),
    ),
    address: PropTypes.string,
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    distanceToClient: PropTypes.number,
    clinicProviderId: PropTypes.number,
  }).isRequired,
};

const mapStateToProps = ({
  heal: {
    location: { latitude, longitude },
  },
}) => ({
  latitude,
  longitude,
});

export default connect(mapStateToProps, {})(ClinicListingItem);
