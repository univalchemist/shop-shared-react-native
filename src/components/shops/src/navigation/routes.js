export const NAVIGATION_HOME_SCREEN_PATH = 'SHOP_HomeScreen';

export const NAVIGATION_HOME_PRODUCT_PATH = 'SHOP_Home_Product';

export const NAVIGATION_HOME_STACK_PATH = 'SHOP_Home';

export const SHOP_SELECT_CATEGORY_MODAL = 'SHOP_SELECT_CATEGORY_MODAL';
export const SHOP_SORT_MODAL = 'SHOP_SORT_SCREEN';
export const SHOP_FILTER_MODAL = 'SHOP_FILTER_SCREEN';

export const SHOP_CART_SCREEN = 'SHOP_CART_SCREEN';
export const SHOP_EDIT_PRODUCT_SCREEN = 'SHOP_EDIT_PRODUCT_SCREEN';
export const SHOP_REVIEW_SCREEN = 'SHOP_REVIEW_SCREEN';
export const SHOP_CHECKOUT_SCREEN = 'SHOP_CHECKOUT_SCREEN';

export const SHOP_DELIVERY_ADDRESSES_SCREEN = 'SHOP_DELIVERY_ADDRESSES_SCREEN';
export const SHOP_ADD_ADDRESS_SCREEN = 'SHOP_ADD_ADDRESS_SCREEN';
export const SHOP_EDIT_ADDRESS_SCREEN = 'SHOP_EDIT_ADDRESS_SCREEN';
export const SHOP_SINGLE_SELECT_MODAL = 'SHOP_SINGLE_SELECT_MODAL';

export const ORDER_HISTORY_SCREEN = 'ORDER_HISTORY_SCREEN';
export const ORDER_HISTORY_DETAIL_SCREEN = 'ORDER_HISTORY_DETAIL_SCREEN';
export const TRACK_ORDER_SCREEN = 'TRACK_ORDER_SCREEN';
export const WRITE_REVIEW_SCREEN = 'WRITE_REVIEW_SCREEN';
export const REVIEW_SUBMITTED_SCREEN = 'REVIEW_SUBMITTED_SCREEN';
