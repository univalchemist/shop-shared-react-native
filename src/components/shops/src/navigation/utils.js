export const getDefaultHeader = theme => {
  return {
    headerStyle: {
      backgroundColor: theme.colors.primary,
    },
    headerTitleStyle: {
      ...theme.typography.titleTextSemiBold,
      alignSelf: 'center',
    },
    headerBackTitle: null,
    headerTintColor: theme.colors.appbarTint,
  };
};
