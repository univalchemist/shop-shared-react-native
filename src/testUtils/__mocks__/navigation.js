const subscriptionMap = new Map();
const navigation = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  goBack: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  reset: jest.fn(),
  setOptions: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  addListener: (state, callback) => {
    const subscriptions = subscriptionMap.get(state) || [];
    subscriptions.push(callback);
    subscriptionMap.set(state, subscriptions);
    return {
      remove: () => {
        const subscriptions = subscriptionMap.get(state);
        subscriptionMap.set(subscriptions.filter(x => x !== callback));
      },
    };
  },

  mockEmitState: state => {
    const subscriptions = subscriptionMap.get(state) || [];
    subscriptions.forEach(x => x());
  },

  clearAll: () => {
    subscriptionMap.clear();
  },
};

export default navigation;
