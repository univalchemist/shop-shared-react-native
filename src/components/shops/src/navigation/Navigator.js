import React from 'react';
import {
  createStackNavigator,
  TransitionPresets,
} from '@react-navigation/stack';
import { connect } from 'react-redux';
import { useIntl } from '@shops/wrappers/core/hooks';
import {
  HomeScreen,
  SelectCategoryScreen,
  FilterByScreen,
  SortByScreen,
  ProductDetailsScreen,
  ReviewScreen,
} from '@shops/screens/home';
import {
  AddAddressScreen,
  DeliveryAddressesScreen,
  EditAddressScreen,
} from '@shops/screens/deliveryAddress';
import { ModalBackButton } from '@shops/wrappers/components';
import { ShopHeader, ClearAllButton } from '@shops/components';
import {
  CartScreen,
  EditProductScreen,
  CheckoutScreen,
} from '@shops/screens/cart';
import { updateSortType, updateFilterTypes } from '@shops/store/actions';

import * as routes from './routes';
import {
  OrderHistoryDetailScreen,
  OrderHistoryScreen,
  ReviewSubmittedScreen,
  TrackOrderScreen,
  WriteReviewScreen,
} from '@shops/screens/orderhistory';
import { SingleSelectModal } from '@wrappers/components/form';

const Stack = createStackNavigator();

const Navigator = ({ updateSortType, updateFilterTypes }) => {
  const intl = useIntl();

  const onClearAllSortByPress = () => {
    updateSortType(null);
  };

  const onClearAllFilterByPress = () => {
    updateFilterTypes({});
  };

  return (
    <Stack.Navigator
      initialRouteName={routes.NAVIGATION_HOME_SCREEN_PATH}
      screenOptions={{ headerBackTitleVisible: false }}
      headerMode="float"
    >
      <Stack.Screen
        name={routes.NAVIGATION_HOME_SCREEN_PATH}
        component={HomeScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.NAVIGATION_HOME_PRODUCT_PATH}
        component={ProductDetailsScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CART_SCREEN}
        component={CartScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_CHECKOUT_SCREEN}
        component={CheckoutScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_PRODUCT_SCREEN}
        component={EditProductScreen}
        options={() => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.editProduct' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_SELECT_CATEGORY_MODAL}
        component={SelectCategoryScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.selectCategory' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_SORT_MODAL}
        component={SortByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.sortBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllSortByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_FILTER_MODAL}
        component={FilterByScreen}
        options={({ navigation }) => ({
          title: intl.formatMessage({ id: 'shop.screenTitle.filterBy' }),
          headerBackImage: () => <ModalBackButton />,
          headerRight: () => (
            <ClearAllButton
              navigation={navigation}
              text={intl.formatMessage({ id: 'shop.common.clearAll' })}
              onPress={onClearAllFilterByPress}
            />
          ),
          ...TransitionPresets.ModalTransition,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_REVIEW_SCREEN}
        component={ReviewScreen}
        options={({ route }) => {
          const ratingCount = route.params.product
            ? route.params.product.ratingsCount
            : 0;
          const title = intl.formatMessage(
            {
              id: 'shop.review.title',
              defaultMessage: `Review(${ratingCount})`,
            },
            { ratingCount },
          );

          return {
            title,
            headerBackImage: () => <ModalBackButton />,
            ...TransitionPresets.ModalTransition,
          };
        }}
      />
      <Stack.Screen
        name={routes.SHOP_DELIVERY_ADDRESSES_SCREEN}
        component={DeliveryAddressesScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_ADD_ADDRESS_SCREEN}
        component={AddAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.addNewAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.SHOP_EDIT_ADDRESS_SCREEN}
        component={EditAddressScreen}
        options={{
          title: intl.formatMessage({ id: 'shop.screenTitle.editAddress' }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        name={routes.ORDER_HISTORY_SCREEN}
        component={OrderHistoryScreen}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={OrderHistoryDetailScreen}
        name={routes.ORDER_HISTORY_DETAIL_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={TrackOrderScreen}
        name={routes.TRACK_ORDER_SCREEN}
        options={{
          title: intl.formatMessage({
            id: 'shop.orderHistory.trackOrderTitle',
          }),
          headerBackImage: () => <ModalBackButton />,
          ...TransitionPresets.ModalTransition,
        }}
      />
      <Stack.Screen
        component={WriteReviewScreen}
        name={routes.WRITE_REVIEW_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
        })}
      />
      <Stack.Screen
        component={ReviewSubmittedScreen}
        name={routes.REVIEW_SUBMITTED_SCREEN}
        options={({ navigation }) => ({
          header: () => <ShopHeader navigation={navigation} />,
          gestureEnabled: false,
        })}
      />
      <Stack.Screen
        name={routes.SHOP_SINGLE_SELECT_MODAL}
        component={SingleSelectModal}
        options={({ route }) => ({
          title: route.params?.title,
          headerBackImage: ModalBackButton,
          ...TransitionPresets.ModalTransition,
        })}
      />
    </Stack.Navigator>
  );
};

export const ShopNavigator = connect(null, {
  updateSortType,
  updateFilterTypes,
})(Navigator);
