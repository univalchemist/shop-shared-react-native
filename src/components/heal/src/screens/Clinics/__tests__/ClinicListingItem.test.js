import React from 'react';
import { renderForTest } from '@testUtils';
import navigation from '@testUtils/__mocks__/navigation';

import ClinicListingItem from '../ClinicListingItem';

const clinicData = {
  id: 336,
  name: 'Clinic Bukit Rimau',
  latitude: 3.1275345,
  longitude: 101.6412337,
  distanceToClient: 11300034.6201797,
  address: 'Ho Chi Minh City, Vietnam',
  doctors: [
    {
      id: 322,
      locale: 'en-HK',
      name: 'Anna Tan',
      gender: 'Female',
      isActive: true,
      clinics: [],
    },
    {
      id: 323,
      locale: 'en-HK',
      name: 'Clinic Bukit Rimau',
      gender: 'Female',
      isActive: true,
      clinics: [],
    },
  ],
};

describe('ClinicListingItem', () => {
  test('Render Clinic Listing Item', async () => {
    const component = renderForTest(<ClinicListingItem clinic={clinicData} />, {
      initialState: {
        heal: { location: { latitude: 0, longitude: 0 } },
      },
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
