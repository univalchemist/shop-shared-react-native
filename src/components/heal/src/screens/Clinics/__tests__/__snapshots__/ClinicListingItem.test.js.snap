// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicListingItem Render Clinic Listing Item 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={false}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <View
                          underlayColor="#f0f0f0"
                        >
                          <View
                            alignItems="center"
                            flexDirection="row"
                            px={4}
                            py={24}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "paddingBottom": 24,
                                  "paddingLeft": 32,
                                  "paddingRight": 32,
                                  "paddingTop": 24,
                                },
                              ]
                            }
                          >
                            <View
                              flex={1}
                              style={
                                Array [
                                  Object {
                                    "flexBasis": 0,
                                    "flexGrow": 1,
                                    "flexShrink": 1,
                                  },
                                ]
                              }
                            >
                              <Text
                                color="#333333"
                                fontFamily="default"
                                fontSize={21}
                                letterSpacing={0.3}
                                lineHeight={26}
                                style={
                                  Array [
                                    Object {
                                      "color": "#333333",
                                      "fontFamily": "Arial",
                                      "fontSize": 21,
                                      "letterSpacing": 0.3,
                                      "lineHeight": 26,
                                    },
                                  ]
                                }
                              >
                                Clinic Bukit Rimau
                              </Text>
                              <Text
                                color="gray.1"
                                fontFamily="default"
                                fontSize={16}
                                letterSpacing={0.3}
                                lineHeight={22}
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "color": "#646464",
                                      "fontFamily": "Arial",
                                      "fontSize": 16,
                                      "letterSpacing": 0.3,
                                      "lineHeight": 22,
                                    },
                                  ]
                                }
                              >
                                Ho Chi Minh City, Vietnam
                              </Text>
                              <View
                                flexDirection="row"
                                marginTop={16}
                                style={
                                  Array [
                                    Object {
                                      "flexDirection": "row",
                                      "marginTop": 16,
                                    },
                                  ]
                                }
                              >
                                <View
                                  flexDirection="row"
                                  style={
                                    Array [
                                      Object {
                                        "flexDirection": "row",
                                      },
                                    ]
                                  }
                                >
                                  <Image
                                    source={
                                      Object {
                                        "testUri": "../../../src/components/heal/images/panel_doctor.png",
                                      }
                                    }
                                    style={
                                      Array [
                                        Object {},
                                      ]
                                    }
                                  />
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={14}
                                    letterSpacing={0.3}
                                    lineHeight={18}
                                    marginLeft={8}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 14,
                                          "letterSpacing": 0.3,
                                          "lineHeight": 18,
                                          "marginLeft": 8,
                                        },
                                      ]
                                    }
                                  >
                                    2 Doctors
                                  </Text>
                                </View>
                              </View>
                            </View>
                            <View
                              marginLeft={3}
                              style={
                                Array [
                                  Object {
                                    "marginLeft": 16,
                                  },
                                ]
                              }
                            >
                              <Image
                                size={30}
                                source={
                                  Object {
                                    "testUri": "../../@cxa-rn/components/src/ListItemWithRightChevron/images/angleRight.png",
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "height": 30,
                                      "width": 30,
                                    },
                                  ]
                                }
                              />
                            </View>
                          </View>
                          <View
                            pl={4}
                            pr={4}
                            style={
                              Array [
                                Object {
                                  "paddingLeft": 32,
                                  "paddingRight": 32,
                                },
                              ]
                            }
                          >
                            <View
                              backgroundColor="gray.5"
                              height={1}
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "#e3e3e3",
                                    "height": 1,
                                    "width": "100%",
                                  },
                                ]
                              }
                              width="100%"
                            />
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
